src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -1190,6 +1192,8 @@ static switch_status_t synth_channel_speak(speech_channel_t *schannel, const cha
 	}
 	if (schannel->state != SPEECH_CHANNEL_PROCESSING) {
 		status = SWITCH_STATUS_FALSE;
+		/* set mrcp channel object to NULL */
+		schannel = (speech_channel_t *) mrcp_application_channel_object_set(schannel->unimrcp_channel, NULL);
```

Comment:
schannel will be NULL here since that method will return a channel->obj which is NULL as well. It will cause a crash since at the end of this function will call schannel->mutex unlock

Reply:
No need to reset the channel object to NULL here since schannel is still alive. If we return a false here is not being checked by FS and will continue calling this function.

Reply:
Removed